### PR TITLE
PHPC-425 and PHPC-426: WriteConcern ctor arg handling and validation

### PR DESCRIPTION
**Pull request by jmikola**
* https://jira.mongodb.org/browse/PHPC-425
 * https://jira.mongodb.org/browse/PHPC-426

### CODE REVIEW

File: src/MongoDB/WriteConcern.c [hunk 1]
```diff
@@ -74,14 +73,21 @@ PHP_METHOD(WriteConcern, __construct)
...
+		mongoc_write_concern_set_w(intern->write_concern, Z_LVAL_P(w));
+	} else if (Z_TYPE_P(w) == IS_STRING) {
+		if (strcmp(Z_STRVAL_P(w), PHONGO_WRITE_CONCERN_W_MAJORITY) == 0) {
 			mongoc_write_concern_set_w(intern->write_concern, MONGOC_WRITE_CONCERN_W_MAJORITY);
 		} else {
-			mongoc_write_concern_set_wtag(intern->write_concern, wstring);
+			mongoc_write_concern_set_wtag(intern->write_concern, Z_STRVAL_P(w));
 		}
+	} else {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected w to be array or object, %s given", zend_get_type_by_const(Z_TYPE_P(w)));
```
Comment (derickr) [inline src/MongoDB/WriteConcern.c]: array or object? You mean: "integer or string" surely?

Comment (jmikola) [inline src/MongoDB/WriteConcern.c]: Surely.

**Comment by jmikola** [comment]
@derickr: Updated, please re-review.

**Comment by derickr** [comment]
LGTM

**Pull request merged**